captum/optim/_utils/image/common.py

```diff
@@ -1,10 +1,66 @@
 import math
 from typing import List, Optional, Tuple
 
+import matplotlib.pyplot as plt
+import numpy as np
 import torch
 
 from captum.optim._utils.reducer import posneg
 
+try:
+    from PIL import Image
+except (ImportError, AssertionError):
+    print("The Pillow/PIL library is required to use Captum's Optim library")
+
+
+def show(
+    x: torch.Tensor, figsize: Optional[Tuple[int, int]] = None, scale: float = 255.0
+) -> None:
+    """
+    Show CHW & NCHW tensors as an image.
+
+    Args:
+        x (torch.Tensor): The tensor you want to display as an image.
+        figsize (Tuple[int, int], optional): height & width to use
+            for displaying the image figure.
+        scale (float): Value to multiply the input tensor by so that
+            it's value range is [0-255] for display.
+    """
+
+    if x.dim() not in [3, 4]:
+        raise ValueError(
+            f"Incompatible number of dimensions. x.dim() = {x.dim()}; should be 3 or 4."
+        )
+    x = torch.cat([t[0] for t in x.split(1)], dim=2) if x.dim() == 4 else x
+    x = x.cpu().detach().permute(1, 2, 0) * scale
```

Comment:
It looks like we are missing a `.clone()` on this line:

```
x = x.cpu().detach().permute(1, 2, 0) * scale
```
It should be:

```
x = x.clone().cpu().detach().permute(1, 2, 0) * scale
```

Reply:
Sounds good let's do that